File: java-workspace3/src/완전탐색/소수찾기.java
package 완전탐색;

import java.util.HashSet;
import java.util.Set;

public class 소수찾기 {
    static int[] idxes;
    static boolean[] visited;
    static Set<Integer> answers = new HashSet<>();

    public static void main(String[] args) {
        solution("17");
    }

    public static int solution(String numbers) {
        idxes = new int[numbers.length()];
        visited = new boolean[numbers.length()];
        comb(numbers, 0);
        return answers.size();
    }

    private static void comb(String numbers, int idx) {
        if (idx == numbers.length()) {
            StringBuilder strNumber = new StringBuilder();
            for (int i = 0; i < numbers.length(); i++) {
                strNumber.append(numbers.charAt(idxes[i]));
            }
            int intNumber = Integer.parseInt(strNumber.toString());
            if(isPrimeNumber(intNumber)){
                answers.add(intNumber);
            }
            return;
        }
        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                idxes[idx] = i;
                comb(numbers, idx + 1);
                visited[i] = false;
            }
        }
    }

    private static boolean isPrimeNumber(int number) {
        if (number == 1) {
            return false;
        }
        for (int num = 2; num < number; num++) {
            if (number % num == 0) {
                return false;
            }
        }
        return true;
    }
}

File: python-workspace/완전탐색/소수찾기.py
from itertools import permutations


def solution(numbers):
    nums = list(str(numbers))
    answers = set()
    for length in range(1, len(numbers) + 1):
        for per in permutations(nums, length):
            number = int("".join(per))
            if is_prime(number):
                answers.add(number)
    return len(answers)


def is_prime(number):
    if number == 1 or number == 0:
        return False
    for num in range(2, number):
        if number % num == 0:
            return False
    return True
